Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm

```diff
@@ -669,25 +669,32 @@ static inline bool shouldAddContentKeyRecipients()
 
     ALWAYS_LOG(LOGIDENTIFIER);
 
-    if (m_cdmInstance && shouldAddContentKeyRecipients()) {
-        if (m_videoLayer)
-            [m_cdmInstance->contentKeySession() removeContentKeyRecipient:m_videoLayer->displayLayer()];
+    if (m_cdmInstance) {
+        if (shouldAddContentKeyRecipients()) {
+            if (m_videoLayer)
+                [m_cdmInstance->contentKeySession() removeContentKeyRecipient:m_videoLayer->displayLayer()];
 
-        for (auto& pair : m_audioRenderers)
-            [m_cdmInstance->contentKeySession() removeContentKeyRecipient:pair.second.get()];
+            for (auto& pair : m_audioRenderers)
+                [m_cdmInstance->contentKeySession() removeContentKeyRecipient:pair.second.get()];
+        }
+        m_cdmInstance->removeKeyStatusesChangedObserver(*m_keyStatusesChangedObserver);
```

Comment:
shouldn't this be done in the destructor too? otherwise we can get a UAF if the cdmInstance receives the keys after the SourceBuffer has been destroyed.

Reply:
Shouldn't be necessary; the CDM stores these in a `WeakHashSet`.